File: Python/8/IsTheStringUppercase/test_is_the_string_uppercase.py
from unittest import TestCase
from unittest import main

from is_the_string_uppercase import is_uppercase


class TestIsTheStringUppercase(TestCase):
    def test_is_upper(self):
        ptr = [
            ('c', False),
            ('C', True),
            ('hello I AM DONALD', False),
            ('HELLO I AM DONALD', True),
        ]
        for inp, exp in ptr:
            with self.subTest(inp=inp, exp=exp):
                self.assertEqual(is_uppercase(inp=inp), exp)


if __name__ == "__main__":
    main(verbosity=2)

File: node/8/IsTheStringUppercase/test_is_the_string_uppercase.js
const assert = require("assert").strict;
var func = require("./is_the_string_uppercase");

let params = {
    "c": false,
    "C": true,
    "hello I AM DONALD": false,
    "HELLO I AM DONALD": true,
    "ACSKLDFJSgSKLDFJSKLDFJ": false,
    "ACSKLDFJSGSKLDFJSKLDFJ": true,
};

// Parameterize Test
Object.keys(params).forEach(function(k) {
    var inp = func(k);
    var exp = params[k];

    assert.equal(inp, exp);
});
